org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/RecordsFilter.java

```diff
@@ -58,6 +60,48 @@ boolean isHashCode(final MethodNode m) {
 			return cursor != null;
 		}
 
+		/**
+		 * Criteria: method name == field name, only three instructions (aload0,
+		 * getField, return), and note that this class only happens in a record,
+		 * so it's safe to assume that this is the record field accessor
+		 * generated. It may happen that the code is explicitly written by the
+		 * developer and is intentionally kept the same as the default generated
+		 * format, but that's just trivial code, and it still makes sense to
+		 * filter them out anyway.
+		 * <p>
+		 * Exception: if the code is compiled within IntelliJ IDEA's Java
+		 * instrumentation, there will be extra null-assertion instructions
+		 * after the getField instruction. This case is <emph>ignored</emph>.
+		 */
+		boolean isFieldAccessor(final MethodNode m) {
+			// No parameter
+			if (!m.desc.startsWith("()")) {
+				return false;
+			}
```

Comment:
Probably a matter of taste, but IMO will be better to move comment into body of the `if`

```suggestion
			if (!m.desc.startsWith("()")) {
				// method with parameters
				return false;
			}
```
or remove it entirely.

Reply:
Done